sql-cli/sql_cli/project.py

```diff
@@ -76,22 +82,34 @@ def airflow_config(self) -> dict[str, Any]:
         parser.read(filename)
         return {section: dict(parser.items(section)) for section in parser.sections()}
 
-    def update_config(self, environment: str = DEFAULT_ENVIRONMENT) -> None:
+    def _initialise_global_config(self) -> None:
         """
-        Sets custom Airflow configuration in case the user is not using the default values.
+        Initialises global config file that includes configuration to be shared across environments including the
+        airflow config.
+        """
+        config = Config(environment=GLOBAL_CONFIG, project_dir=self.directory)
+        global_env_filepath = config.get_global_config_filepath()
+        config.write_value_to_yaml("airflow", "home", str(self._airflow_home.resolve()), global_env_filepath)
+        if not Path.exists(self._airflow_dags_folder):
+            raise FileNotFoundError(f"Specified DAGs directory {self._airflow_dags_folder} does not exist.")
+        config.write_value_to_yaml(
+            "airflow", "dags_folder", str(self._airflow_dags_folder.resolve()), global_env_filepath
+        )
 
-        :param airflow_home: Custom user-defined Airflow Home directory
-        :param airflow_dags_folder: Custom user-defined Airflow DAGs folder
+    def transform_env_config(self, environment: str = DEFAULT_ENVIRONMENT) -> None:
         """
-        config = Config(environment=environment, project_dir=self.directory)
-        config = config.from_yaml_to_config()
+        Transforms environment specific configurations post project initialisation.
 
-        if self._airflow_home is not None:
-            config.write_value_to_yaml("airflow", "home", str(self._airflow_home.resolve()))
-        if self._airflow_dags_folder is not None:
-            config.write_value_to_yaml("airflow", "dags_folder", str(self._airflow_dags_folder.resolve()))
+        E.g. the example workflows have relative paths for the host URLs for SQLite connections. They need to be
+        converted to absolute paths once the project directory is initialised so that the connections work successfully.
 
-        config.connections[0]["host"] = str(self.directory / config.connections[0]["host"])
+        :param environment: the environment for which the configuration has to be updated
+        """
+        config = Config(environment=environment, project_dir=self.directory)
+        config = config.from_yaml_to_config()
+        for connection in config.connections:
+            if connection["conn_type"] == SQLITE_CONN_TYPE and not os.path.isabs(connection["host"]):
+                connection["host"] = str(self.directory / connection["host"])
```

Comment:
This seems to be quite specific. Could we move that into its own context/function? What does this do? If it makes the path absolute we can use pathlib .resolve to make it absolute.

Reply:
Resolve does not help as it somehow resolves it to the directory from where the command is called and does not necessary prefix the path with the initialised project's path. We can move it to a separate function once we have more use cases of what all transformations need to be done. Right now the function is quite small enough.

Reply:
As discussed over call, we are also checking that the resolved path exists. If it does not exist, we raise an exception. It is expected that the user specifies an absolute path. So, if a user has specified an absolute path, the code is a no-operation. But, if it is not an absolute path, we try to resolve it with the project directory as the base path and if the resolved path does not exist we raise an exception. I think this is the best we can do (Do nothing if it is an absolute path. If not try the best to resolve by prefixing with project directory).